src/main/java/seedu/sudowudo/model/item/Item.java

```diff
@@ -263,42 +256,45 @@ public String toString() {
         return description.toString();
     }
 
-
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof ReadOnlyItem // instanceof handles nulls
-                && this.isSameStateAs((ReadOnlyItem) other));
+                        && this.isSameStateAs((ReadOnlyItem) other));
     }
 
-    //@@author A0147609X
+    // @@author A0147609X
     @Override
     /**
      * sort by start date then end date then alphabetically
      * for UI chronological sort
      * @author darren
      */
     public int compareTo(Item other) {
-        LocalDateTime thisStart, thisEnd, otherStart, otherEnd;
-        
-        thisStart = assignDummyLDT(startDate);
-        thisEnd = assignDummyLDT(endDate);
-        otherStart = assignDummyLDT(other.getStartDate());
-        otherEnd = assignDummyLDT(other.getEndDate());
+        LocalDateTime thisStart = assignDummyLDT(this.startDate);
+        LocalDateTime thisEnd = assignDummyLDT(this.endDate);
+        LocalDateTime otherStart = assignDummyLDT(other.getStartDate());
+        LocalDateTime otherEnd = assignDummyLDT(other.getEndDate());
         
-        if(thisStart.isBefore(otherStart)) {
+        // Assign same start/end date to a deadline for easier checking
+        if (this.is(Type.TASK.toString())){
```

Comment:
Override the `toString` method in the enum `Type` (inside `Item.java`) so that it is not necessary to call `toString()` here for cleanliness' sake.

Change should be:
```java
public enum Type {
        // blah blah some important stuff here

        /* let's get rid of this method
        public String getTypeName() {
            return this.typeName;
        }
        */

        // and put this one in its place
        @Override
        public String toString() {
            return this.typeName;
        }
    }
```

The new condition in this line would be
```java
if (this.is(Type.TASK)) {
```